### PR TITLE
Elastic synhronise projects

### PR DESCRIPTION
**Pull request by maxhodik**

### CODE REVIEW

File: elastic/src/main/java/com/hodik/elastic/controller/ProjectController.java [hunk 1]
```diff
@@ -37,6 +37,12 @@ public ResponseEntity<HttpStatus> createProject(@RequestBody ProjectDto projectD
         return ResponseEntity.ok(HttpStatus.OK);
     }
 
+    @PostMapping("/sync")
+    public ResponseEntity<HttpStatus> createProjectList(@RequestBody List<ProjectDto> projectDtoList) {
+        projectService.createProjectList(projectDtoList.stream().map(projectMapper::convertToProject).toList());
```
Comment (Gtm417) [inline elastic/src/main/java/com/hodik/elastic/controller/ProjectController.java]: extract stream result to variable

File: elastic/src/main/java/com/hodik/elastic/controller/ProjectController.java [hunk 2]
```diff
@@ -37,6 +37,12 @@ public ResponseEntity<HttpStatus> createProject(@RequestBody ProjectDto projectD
         return ResponseEntity.ok(HttpStatus.OK);
     }
 
+    @PostMapping("/sync")
+    public ResponseEntity<HttpStatus> createProjectList(@RequestBody List<ProjectDto> projectDtoList) {
```
Comment (Gtm417) [inline elastic/src/main/java/com/hodik/elastic/controller/ProjectController.java]: why does it called createProjectList?

File: elastic/src/main/java/com/hodik/elastic/dto/ProjectDto.java [hunk 3]
```diff
@@ -22,34 +25,29 @@ public class ProjectDto {
...
-    @JsonFormat(pattern = "dd/MM/yyyy")
-    private LocalDate createdDate; //(timestamp)
-    @JsonFormat(pattern = "dd/MM/yyyy")
-    private LocalDate finalPlannedDate;// (timestamp) //– date when project ends
-    @JsonFormat(pattern = "dd/MM/yyyy")
-    private LocalDate startDate; //(timestamp) //– date when project starts
+
+    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm")
+    @JsonDeserialize(using = CustomLocalDateTimeDeSerializer.class)
+    private LocalDateTime createdDate; //(timestamp)
```
Comment (Gtm417) [inline elastic/src/main/java/com/hodik/elastic/dto/ProjectDto.java]: remove comments

File: elastic/src/main/java/com/hodik/elastic/dto/ProjectDto.java [hunk 4]
```diff
@@ -22,34 +25,29 @@ public class ProjectDto {
...
-                .append("isPrivate", isPrivate)
-                .append("createdDate", createdDate)
-                .append("finalPlannedDate", finalPlannedDate)
-                .append("startDate", startDate)
-                .append("category", category)
-                .append("description", description)
-                .append("isCommercial", isCommercial)
-                .append("status", status)
-                .toString();
+        return new ToStringBuilder(this, ToStringStyle.JSON_STYLE).append("id", id).append("name", name).append("isPrivate", isPrivate).append("createdDate", createdDate).append("finalPlannedDate", finalPlannedDate).append("startDate", startDate).append("category", category).append("description", description).append("isCommercial", isCommercial).append("status", status).toString();
```
Comment (Gtm417) [inline elastic/src/main/java/com/hodik/elastic/dto/ProjectDto.java]: make it not in one line

File: elastic/src/main/java/com/hodik/elastic/model/Project.java [hunk 5]
```diff
@@ -26,23 +27,26 @@ public class Project {
...
     private String name;
     @Field(type = FieldType.Boolean)
-    private boolean isPrivate;
+    private Boolean isPrivate;
+
     @Field(type = FieldType.Date)
-    @JsonFormat(pattern = "dd/MM/yyyy")
-    private LocalDate createdDate; //(timestamp)
+    @ValueConverter(CustomDatePropertyValueConverter.class)
+    private LocalDateTime createdDate; //(timestamp)
```
Comment (Gtm417) [inline elastic/src/main/java/com/hodik/elastic/model/Project.java]: remove comments

File: projects_app/src/main/java/org/example/projectapp/controller/ProjectController.java [hunk 6]
```diff
@@ -54,6 +58,11 @@ public ResponseEntity<?> makeProjectPrivate(@PathVariable("id") Long id,
         return ResponseEntity.ok(projectResponseDto);
     }
 
+    @GetMapping()
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/controller/ProjectController.java]: remove unneeded brackets

File: projects_app/src/main/java/org/example/projectapp/controller/RebaseProjectsToElasticController.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    public RebaseProjectsToElasticController(ProjectService projectService, ProjectMapper projectMapper, ElasticProjectsServiceClient elasticProjectsServiceClient) {
+        this.projectService = projectService;
+        this.projectMapper = projectMapper;
+        this.elasticProjectsServiceClient = elasticProjectsServiceClient;
+    }
+
+
+    @PostMapping("/sync")
+    public HttpEntity<HttpStatus> rebaseProjects(@RequestBody @Nullable List<Long> ids) {
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/controller/RebaseProjectsToElasticController.java]: why rebase?

File: projects_app/src/main/java/org/example/projectapp/mapper/dto/ProjectElasticDto.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+import org.example.projectapp.model.ProjectStatus;
+import org.example.projectapp.util.CustomLocalDateTimeSerializer;
+
+import java.time.LocalDateTime;
+
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/mapper/dto/ProjectElasticDto.java:19]: do you use all args constructor?

File: projects_app/src/main/java/org/example/projectapp/restclient/ElasticProjectsServiceClient.java [hunk 9]
```diff
@@ -1,13 +1,26 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
 
 import java.util.List;
 
-@FeignClient(name = "${service.elastic.feign.name}", url = "${service.elastic.feign.url")
+@FeignClient(name = "${service.elastic.projects.feign.name}", url = "${service.elastic.feign.url}/projects")
 public interface ElasticProjectsServiceClient {
-    @GetMapping("/projects")
+    @GetMapping()
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/restclient/ElasticProjectsServiceClient.java]: brackets

File: projects_app/src/main/java/org/example/projectapp/restclient/ElasticUsersServiceClient.java [hunk 10]
```diff
@@ -10,17 +10,17 @@
 
 import java.util.List;
 
-@FeignClient(name = "${service.elastic.feign.name}", url = "${service.elastic.feign.url}")
+@FeignClient(name = "${service.elastic.feign.name}", url = "${service.elastic.feign.url}/users")
 public interface ElasticUsersServiceClient {
-    @GetMapping("/users")
+    @GetMapping()
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/restclient/ElasticUsersServiceClient.java]: brackets

File: projects_app/src/main/java/org/example/projectapp/service/impl/ProjectServiceImpl.java [hunk 11]
```diff
@@ -56,6 +63,7 @@ public Project createProject(ProjectDto projectDto) {
         User userFromAuth = authService.getUserFromAuth();
         projectRepository.saveAndFlush(project);
         projectMemberService.saveProjectMemberAndReturn(ProjectRole.OWNER, project, userFromAuth);
+        elasticProjectsServiceClient.createProject(projectMapper.convertToProjectElasticDto(project));
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/service/impl/ProjectServiceImpl.java]: extract mapper executions to variable

File: projects_app/src/main/java/org/example/projectapp/service/impl/ProjectServiceImpl.java [hunk 12]
```diff
@@ -77,12 +85,13 @@ public void enableNotification(Long projectId, boolean enable) {
     public ProjectResponseDto updateProjectInfo(Long id, ProjectInfoDto dto) {
         Project project = tryGetProject(id);
         mergeProject(project, dto);
-
+        elasticProjectsServiceClient.updateProject(id, projectMapper.convertToProjectElasticDto(project));
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/service/impl/ProjectServiceImpl.java]: same with mapper

File: projects_app/src/main/java/org/example/projectapp/service/impl/ProjectServiceImpl.java [hunk 13]
```diff
@@ -92,7 +101,7 @@ private ProjectResponseDto saveAndReturnDto(Project project) {
     public ProjectResponseDto makeProjectPrivate(Long id, Boolean isPrivate) {
         Project project = tryGetProject(id);
         project.setPrivate(isPrivate);
-
+        elasticProjectsServiceClient.updateProject(id, projectMapper.convertToProjectElasticDto(project));
```
Comment (Gtm417) [inline projects_app/src/main/java/org/example/projectapp/service/impl/ProjectServiceImpl.java]: same

File: projects_app/src/main/java/org/example/projectapp/mapper/dto/ProjectElasticDto.java (hunk 8 above)
Comment (maxhodik) [reply projects_app/src/main/java/org/example/projectapp/mapper/dto/ProjectElasticDto.java:19]: it needs for Bilder

File: elastic/es-scripts/projects-index-create [hunk 14]
```diff
@@ -3,41 +3,39 @@ curl --location --request PUT 'localhost:9200/projects' \
 --header 'Content-Type: application/json' \
 --header 'Authorization: Basic ZWxhc3RpYzplbGFzdGlj' \
```
Comment (Gtm417) [inline elastic/es-scripts/projects-index-create]: remove auth from every file

File: projects_app/src/main/resources/application.properties [hunk 15]
```diff
@@ -1,7 +1,7 @@
 spring.datasource.url=jdbc:mysql://localhost:3306/stud_projects?useUnicode=true&serverTimezone=UTC
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
 spring.datasource.username=root
-spring.datasource.password=root
+spring.datasource.password=1234
```
Comment (Gtm417) [inline projects_app/src/main/resources/application.properties]: remove

**Pull request merged**